src/scheduler/multi_threading.cpp

```diff
@@ -240,7 +238,7 @@ worker(void *tid)
 	ntid = *(int *)tid;
 
 	/* Add HUP to the list of signals to block, if we ever unblock this, we'll need to modify 'restart()' to handle MT */
-	set = allsigs;
+	pthread_sigmask(SIG_BLOCK, NULL, &set);
```

Comment:
Just a minor comment, shouldn't this be SIG_SETMASK? something like pthread_sigmask(SIG_SETMASK, &set, NULL)?

Reply:
As we discussed Arun, when 'set' is NULL, the first argument is ignored, so it doesn't really matter.

Reply:
Nevermind, I was confused by how man page has explained SIG_SETMASK.  https://www.gnu.org/software/libc/manual/html_node/Process-Signal-Mask.html explains better. Please ignore the comment.

Reply:
I think u could do this in one pthread_sigmask() call right? SIG_BLOCK is supposed to "add" to the current mask-set
By doing this:
sigemptyset(&set);
sigaddset(&set, SIGHUP);
pthread_sigmask(SIG_BLOCK, &set, NULL);